Extend the documentation with more information about multidimensional ranges (#1569)
uxlfoundation/oneTBB: doc/main/tbb_userguide/Advanced_Topic_Other_Kinds_of_Iteration_Spaces.rst

loop to be "recursively blocked" in a way that improves cache usage.
This nice cache behavior means that using ``parallel_for`` over a
``blocked_range2d<T>`` can make a loop run faster than the sequential
equivalent, even on a single processor. 

> Also, ``blocked_range2d`` allows to use different value types across

Reviewer: The ``blocked_range2d`` allows you to use different value types for its two dimensions: *rows* (the first dimension) and *columns* (the second dimension).

its first dimension (called "rows") and the second one ("columns").

> That allows combining indexes, pointers, and iterators into a joint

Reviewer: ```suggestion
That means you can combine indexes, pointers, and iterators into a joint
```

> iteration space. The method functions ``rows()`` and ``cols()`` return

Reviewer: To get the range for each dimension, use the ``rows()`` and ``cols()`` methods that return the respective dimensions as ``blocked_range`` objects.

corresponding dimensions in the form of a ``blocked_range``.

The ``blocked_range3d`` class template extends this approach to 3D by adding
``pages()`` as the first dimension, followed by ``rows()`` and ``cols()``.

> The ``blocked_nd_range<T,N>`` class template represents a blocked iteration

Reviewer: I'm not sure that I got it right, but maybe smth like:

The ``blocked_nd_range<T,N>`` class template represents a blocked iteration space of any dimensionality. However, unlike other classes, all dimensions in ``blocked_nd_range`` must share the same value type. Instead of specifying boundary values directly, you pass ``N`` instances of ``blocked_range<T>`` to the constructor. This difference is also reflected in its different naming pattern.

space of any dimensionality, but in a slightly different way. All dimensions
of ``blocked_nd_range`` must be specified over the same value type, and the
constructor takes N instances of ``blocked_range<T>``, not individual boundary
values. To indicate the distinctions, the different naming pattern was chosen.

> An Example of a Multidimensional Iteration Space

Reviewer: ```suggestion
Example of a Multidimensional Iteration Space
```

------------------------------------------------

The example demonstrates calculation of a 3-dimensional filter over the pack

> of feature maps, applying a kernel to a subrange of features.

Reviewer: ```suggestion
of feature maps by applying a kernel to a subrange of features.
```

> The ``convolution3d`` function iterates over the output cells and sets cell

Reviewer: The ``convolution3d`` function processes each output cell, setting its value based on the result of the ``kernel3d`` function. This function combines values from the feature maps to produce the final result.

values to the result of the ``kernel3d`` function, which summarizes values
from feature maps.

> For the computation to be performed in parallel, ``tbb::parallel_for`` is called

Reviewer: ```suggestion
To enable parallel computation, ``tbb::parallel_for`` is called
```

with ``tbb::blocked_nd_range<int,3>`` as an argument. The body function then

> iterates over the received 3-dimensional subrange in a loop nest, using

Reviewer: Within the body function, a nested loop iterates over the 3D subrange received. The ``dim`` method is used to get the loop boundaries for each dimension.
